README.md

```diff
@@ -43,118 +55,524 @@ Check the original technical report for an earlier version of the Model Extensio
+
+> [!NOTE]
+> (*) These roles are offered as direct conversions from the previous extension
+> that provided [ML-Model Asset Roles][ml-model-asset-roles] to provide easier upgrade to the MLM extension.
+
+[ml-model-asset-roles]: https://github.com/stac-extensions/ml-model?tab=readme-ov-file#asset-objects
+
+### Model Asset
+
+| Field Name        | Type                                      | Description                                                                                      |
+|-------------------|-------------------------------------------|--------------------------------------------------------------------------------------------------|
+| title             | string                                    | Description of the model asset.                                                                  |
+| href              | string                                    | URI to the model artifact.                                                                       |
+| type              | string                                    | The media type of the artifact (see [Model Artifact Media-Type](#model-artifact-media-type).     |
+| roles             | \[string]                                 | **REQUIRED** Specify `mlm:model`. Can include `["mlm:weights", "mlm:checkpoint"]` as applicable. |
+| mlm:artifact_type | [Artifact Type Enum](#artifact-type-enum) | Specifies the kind of model artifact. Typically related to a particular ML framework.            |
+
+Recommended Asset `roles` include `mlm:weights` or `mlm:checkpoint` for model weights that need to be loaded by a
+model definition and `mlm:compiled` for models that can be loaded directly without an intermediate model definition.
+In each case, the `mlm:model` should be applied as well to indicate that this asset represents the model.
+
+It is also recommended to make use of the
+[file](https://github.com/stac-extensions/file?tab=readme-ov-file#asset--link-object-fields)
+extension for this Asset, as it can provide useful information to validate the contents of the model definition,
+by comparison with fields `file:checksum` and `file:size` for example.
+
+#### Model Artifact Media-Type
+
+Not all ML framework, libraries or model artifacts provide explicit media-type. When those are not provided, custom
+media-types can be considered. For example `application/x-pytorch` or `application/octet-stream; application=pytorch`
+could be appropriate to represent a PyTorch `.pt` file, since the underlying format is a serialized pickle structure.
+
+#### Artifact Type Enum
+
+This value can be used to provide additional details about the specific model artifact being described.
+For example, PyTorch offers various strategies for providing model definitions, such as Pickle (`.pt`), TorchScript,
+or the compiled approach. Since they all refer to the same ML framework,
```

Comment:
```suggestion
or the upcoming [Ahead-Of-Time Compiled .pt2 format](https://pytorch.org/docs/main/torch.compiler_aot_inductor.html). Since they all refer to the same ML framework,
```